usaspending_api/search/v2/views/spending_by_award_count.py

```diff
@@ -0,0 +1,77 @@
+import copy
+
+from django.conf import settings
+from django.db.models import Count
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from usaspending_api.awards.v2.filters.matview_filters import matview_search_filter
+from usaspending_api.awards.v2.filters.sub_award import subaward_filter
+from usaspending_api.awards.v2.lookups.lookups import all_awards_types_to_category, assistance_type_mapping
+from usaspending_api.common.api_versioning import api_transformations, API_TRANSFORM_FUNCTIONS
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.exceptions import InvalidParameterException
+from usaspending_api.common.helpers.orm_helpers import category_to_award_materialized_views
+from usaspending_api.common.validator.award_filter import AWARD_FILTER
+from usaspending_api.common.validator.pagination import PAGINATION
+from usaspending_api.common.validator.tinyshield import TinyShield
+
+
+@api_transformations(api_version=settings.API_VERSION, function_list=API_TRANSFORM_FUNCTIONS)
+class SpendingByAwardCountVisualizationViewSet(APIView):
```

Comment:
So the biggest difference I see here is that the existing code uses summary views to get counts when it can whereas this is trying to count from the materialized views directly.  I believe this pretty much single-handedly explains the performance difference on counts.

Just trying to count `mv_search_award_contract` by itself is taking 1 minute on staging for FY2019.

Reply:
Close. There was only one "summary award" materialized view, and it was a small subset of filters. TAS would have used universal_award_matview.

I know the current performance sucks, I foolishly switched back to the Django ORM without proper performance tests after the fact

Reply:
Re-implemented the async SQL calls. Performance is much better during my tests

Reply:
Re-implemented the async SQL calls. Performance is much better during my tests